.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,43 @@
+name: Run google-benchmark pre-commit checks
```

Comment:
nit: all of these actions are pre-commit checks.  this is specific to python, unless we include the clang format stuff here too.  can you name it appropriately please?

Reply:
Should I rename, or look for a clang-format hook and try to include that in this job?

Reply:
i think just rename.  i want the clang format to be required but this likely isn't.

Reply:
Done, name still tentative (it doesn't really fit the currently existing names tbh).

Unsurprisingly, there are quite some reformatting suggestions, so I will defer to you in how I should tackle them.

Reply:
> i think just rename. i want the clang format to be required but this likely isn't.

fwiw, GitHub says that they are scheduling required workflows for deprecation here: https://docs.github.com/en/actions/using-workflows/required-workflows

Seems like they just assume you to never merge in case any workflow you want isn't green.

Reply:
> Done, name still tentative (it doesn't really fit the currently existing names tbh).
> 
> Unsurprisingly, there are quite some reformatting suggestions, so I will defer to you in how I should tackle them.
> 
> First complete job: https://github.com/google/benchmark/actions/runs/6666832134/job/18119132126

i don't think we should have a CI bot that we know is failing (yes i know about the sanitizer ones. it's on my list) so maybe we need to fix these first?

Reply:
Sure, no problem. Should I put them as blame ignore-revs to not mess up the blame view? (That would require you to not squash-merge them)

Source: https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view

Reply:
no strong preference.